### DISCUSSION TITLE
Default shell?

### DISCUSSION THREAD

**Discussion by rrveex** [sxyazi/yazi#718] [1. Q&A]
Seems like `shell` commands always uses `sh`. This leads to for instance not even built-in plugins work correctly, like `archive` - when I do `unpack here`, it says
`read: arg count` and closes the shell automatically (I have to open it again with for instance `shell zsh --block` to be able to see the error message).

Could we have a "default shell" setting? (instead of the hard-coded `Command::new("sh")` in `shell.rs`)

I know, I could do something like `"shell 'zsh -c "read; whatever;"'"`... but it's kind of ugly I think... probably have to escape the '"'"

**Comment by sxyazi** [comment]
In case someone passes by, we discussed it at https://github.com/sxyazi/yazi/pull/742
